src/Microsoft.AspNetCore.Sockets.Client/DefaultTransportFactory.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.Extensions.Logging;
+using System;
+using System.Net.Http;
+
+namespace Microsoft.AspNetCore.Sockets.Client
+{
+    public class DefaultTransportFactory : ITransportFactory
+    {
+        private readonly TransportType _requestedTransportType;
+
+        public DefaultTransportFactory(TransportType requestedTransportType)
+        {
+            if (requestedTransportType <= 0 || requestedTransportType > TransportType.All)
+            {
+                throw new ArgumentOutOfRangeException(nameof(requestedTransportType));
+            }
+
+            _requestedTransportType = requestedTransportType;
+        }
+
+        public ITransport CreateTransport(TransportType availableServerTransports, ILoggerFactory loggerFactory, HttpClient httpClient)
```

Comment:
No. Note that this code is not being called by the user. This is user's code being invoked from SignalR and we provide you all the things we might create and you might need to create the transport. You don't have to use them as long as you give us a transport (or throw)